nsqd/client_v2.go

```diff
@@ -499,6 +503,7 @@ func (c *clientV2) UpgradeTLS() error {
 	c.Writer = bufio.NewWriterSize(c.tlsConn, c.OutputBufferSize)
 
 	atomic.StoreInt32(&c.TLS, 1)
+	atomic.StoreInt32(&c.ClientAuth.TLS, 1)
```

Comment:
Agreed. I think `TLS` should live in `clientV2` and be passed to the auth functions.